src/rlookup.c

```diff
@@ -625,23 +625,20 @@ static int getKeyCommonHash(const RLookupKey *kk, RLookupRow *dst, RLookupLoadOp
   }
 
   // Get the actual hash value
-  int rc = REDISMODULE_ERR;
   RedisModuleString *val = NULL;
   RSValue *rsv = NULL;
 
-  rc = RedisModule_HashGet(*keyobj, REDISMODULE_HASH_CFIELDS, kk->path, &val, NULL);
+  RedisModule_HashGet(*keyobj, REDISMODULE_HASH_CFIELDS, kk->path, &val, NULL);
 
-  if (rc == REDISMODULE_OK && val != NULL) {
+  if (val != NULL) {
     // `val` was created by `RedisModule_HashGet` and is owned by us.
     // This function might retain it, but it's thread-safe to free it afterwards without any locks
     // as it will hold the only reference to it after the next line.
     rsv = hvalToValue(val, (kk->flags & RLOOKUP_T_NUMERIC) ? RLOOKUP_C_DBL : RLOOKUP_C_STR);
     RedisModule_FreeString(RSDummyContext, val);
-  } else if (!strncmp(kk->path, UNDERSCORE_KEY, strlen(UNDERSCORE_KEY))) {
-    RedisModuleString *keyName = RedisModule_CreateString(options->sctx->redisCtx,
-                                  keyPtr, strlen(keyPtr));
+  } else if (!strcmp(kk->path, UNDERSCORE_KEY)) {
+    const RedisModuleString *keyName = RedisModule_GetKeyNameFromModuleKey(*keyobj);
```

Comment:
which API version support this?

Reply:
6.0.0 (according to [the docs](https://redis.io/docs/reference/modules/modules-api-ref/#RedisModule_GetKeyNameFromModuleKey))